### PR TITLE
Removed BIB file directory from search when preferences has option unchecked

### PR DESCRIPTION
**Pull request by silverhorse**
Fixes https://github.com/JabRef/jabref/issues/5891 
For "Automatically set file links" feature, the BIB file directory is always added for search, even though the checkbox is not clicked. If the checkbox is clicked, it is added at first position in list of directories to search, which is expected behaviour. However, it is still added to list of directories to search even when checkbox is left unselected, which is faulty. I fixed this by removing code which adds the directory when preference is set to false.

Also, changed the text of the file preferences in order to make it clear that files will be searched only when the checkbox "Search for files relative to BIB file location" is selected. Also changing user documentation to reflect this text change - https://github.com/JabRef/user-documentation/pull/283 

Before:
Screen Shot 2020-05-26 at 3 44 34 PM

After:
preferences-file-usethebibfilelocationasprimaryfiledirectory

- [x] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
Note to reviewers: We should really carefully think about that. Refs https://github.com/JabRef/jabref/pull/459 (and linked PRs from there).

**Comment by silverhorse** [comment]
> Note to reviewers: We should really carefully think about that. Refs #459 (and linked PRs from there).

Yes, I feel there is something amiss. Because when there is no directory selected in meta-data, user provided dir or BIB default directory, then the search directory list will be empty!
May be we should provide another option where the primary dir is removed only explicitly by user.

**Comment by koppor** [comment]
We certainly need to check the documentation. Does it cover all cases discussed in the linked PR? Is the documetnation enough to serve as specification?

JabRef has very powerful diretory features and we should be careful when touching that code.

**Comment by silverhorse** [comment]
> We certainly need to check the documentation. Does it cover all cases discussed in the linked PR? Is the documetnation enough to serve as specification?
> 
> JabRef has very powerful diretory features and we should be careful when touching that code.


No, I have to look at the linked PR and check for all cases. I will get to it in a bit. Certainly, its not taken lightly. Will check for all cases.

**Comment by Siedlerchr** [comment]
So, if all three directories are empty (main file dir and the two in library properties and "Use bib file location as primary dir is checked" ) then it would return now an empty collection. 
Previously it always returned the bib file location as last resort.

And when I check the checkbox "Use bib file location" the bib file location is returned as first location.

@silverhorse  It would be nice if you add a test case for the Move Files Cleanup formatter which tests with all directories empty and the checkbox not checked. It should probably return false or  does nothing

**Comment by silverhorse** [comment]
@Siedlerchr yes will add a testcase for all dirs empty. Thanks!

**Comment by silverhorse** [comment]
Tested with all combinations of the four options -
1. Library properties - general dir
2. Library properties - user specified dir
3. Preferences - main file directory
4. Preferences - BIB file dir

**Comment by silverhorse** [comment]
@koppor  and @Siedlerchr plz review when you can. I have added some tests and done manual testing. Plz let me know if there are other scenarios I should test apart from one mentioned in msgs.

File: src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java [hunk 1]
```diff
@@ -30,24 +30,33 @@
...
-        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(Collections.singletonList(path.getParent()));
-        when(externalFileTypes.getExternalFileTypeSelection()).thenReturn(new TreeSet<>(ExternalFileTypes.getDefaultExternalFileTypes()));
     }
 
     @Test
-    public void test() throws Exception {
+    public void testFindAssociatedNotLinkedFilesSuccess() throws Exception {
         // Due to mocking the externalFileType class, the file extension will not be found
 
+        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(Collections.singletonList(path.getParent()));
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java:46]: Why did you move the mockings here?

Comment (silverhorse) [reply src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java:46]: This mocking of getFileDirectoriesAsPaths method returns non empty values, which is not applicable for the newly added test case testFindAssociatedNotLinkedFilesForEmptySearchDir. The new test case expects that getFileDirectoriesAsPaths would return empty list.

Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java:46]: Okay, but the externalFileType should be independent of this

Comment (silverhorse) [reply src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java:46]: yeah, agreed. I will move that one back to setup.

**Comment by silverhorse** [comment]
Thanks for guidance and approval @Siedlerchr . Do you think more reviewers are needed for this change or is it good to merge?

**Comment by calixtus** [comment]
> Thanks for guidance and approval @Siedlerchr . Do you think more reviewers are needed for this change or is it good to merge?

We have a strict rule in JabRef that each PR must be reviewed by at least two core developers before it can be merged with the master branch. Be patient. JabRef will still be there tomorrow and many other issues that need to be fixed. :wink:

**Comment by silverhorse** [comment]
@tobiasdiez yes, I like your suggestion. It will remove any confusion regarding this option. Will make this change.

**Comment by abadar** [comment]
@tobiasdiez I am working further to help @silverhorse, here is my change:
https://github.com/silverhorse/jabref/pull/1

I have question regarding localization, is there any step on how to get the translated text? or it should be automatic? I saw its recommended to use Crowdin but didn't find specific information on how the strings will be translated. can you please help out

**Comment by tobiasdiez** [comment]
As a developer you don't have to worry about the actual translation. Simply use `Localization.lang(...)` to get the localizated text and add an entry to `JabRef_en`. After the PR is merged, the text automatically appears on Crowdin and people can translate it there (which is then synced back to JabRef).
Some information can be found at https://devdocs.jabref.org/getting-into-the-code/code-howtos#using-localization-correctly, but this is a bit out-dated. It would be nice if you could update this documentation.

**Comment by abadar** [comment]
updated user documentation: https://github.com/JabRef/user-documentation/pull/283

File: src/main/java/org/jabref/gui/preferences/FileTab.fxml [hunk 2]
```diff
@@ -43,7 +43,7 @@
         <TextField fx:id="mainFileDir" HBox.hgrow="ALWAYS"/>
         <Button onAction="#mainFileDirBrowse" text="%Browse"/>
     </HBox>
-    <CheckBox fx:id="useBibLocationAsPrimary" text="%Use the BIB file location as primary file directory">
+    <CheckBox fx:id="useBibLocationAsPrimary" text="%Search for files relative to BIB file location">
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/preferences/FileTab.fxml:46]:
```suggestion
    <CheckBox fx:id="useBibLocationAsPrimary" text="%Search and store files relative to BIB file location">
```

Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/preferences/FileTab.fxml:46]: and maybe replace "bib file" with "library file" to be consistent with the more recent usage

Comment (abadar) [reply src/main/java/org/jabref/gui/preferences/FileTab.fxml]: done

**Comment by Siedlerchr** [comment]
Thanks for your contribution and the quick follow up! We hope you enjoyed contributing and we are looing forward for more PRs from you  🐱

**Comment by silverhorse** [comment]
Yay! Thanks @Siedlerchr and @tobiasdiez , appreciate your help! We are excited for our first OSS contribution!  Thanks @abadar for last mile help. 🤩
